ui-v2/app/services/repository/service.js

```diff
@@ -7,16 +7,35 @@ export default RepositoryService.extend({
   },
   findBySlug: function(slug, dc) {
     return this._super(...arguments).then(function(item) {
-        const nodes = get(item, 'Nodes');
-        const service = get(nodes, 'firstObject');
-        const tags = nodes
-          .reduce(function(prev, item) {
-            return prev.concat(get(item, 'Service.Tags') || []);
-          }, [])
-          .uniq();
-        set(service, 'Tags', tags);
-        set(service, 'Nodes', nodes);
-        return service;
+      const nodes = get(item, 'Nodes');
+      const service = get(nodes, 'firstObject');
+      const tags = nodes
+        .reduce(function(prev, item) {
+          return prev.concat(get(item, 'Service.Tags') || []);
+        }, [])
+        .uniq();
+      set(service, 'Tags', tags);
+      set(service, 'Nodes', nodes);
+      return service;
+    });
+  },
+  findInstanceBySlug: function(id, slug, dc, configuration) {
+    return this.findBySlug(slug, dc, configuration).then(function(item) {
+      const i = item.Nodes.findIndex(function(item) {
+        return item.Service.ID === id;
       });
+      if (i !== -1) {
+        const service = item.Nodes[i].Service;
+        service.Node = item.Nodes[i].Node;
+        service.ServiceChecks = item.Nodes[i].Checks.filter(function(item) {
+          return item.ServiceID != '';
+        });
+        service.NodeChecks = item.Nodes[i].Checks.filter(function(item) {
+          return item.ServiceID == '';
+        });
+        return service;
+      }
+      // TODO: probably need to throw a 404 here?
```

Comment:
We'll probably look to add 404 checking after this goes in. It's probably unlikely here, and as such will definitely happen at some point 😁 

We plan on adding a lot more testing ontop of this, so this will probably happen in tandem with that.